### PR TITLE
Avoid using v_signmask universal intrinsic and mark it as deprecated

### PR DESCRIPTION
**Pull request by terfendail**
### This pullrequest changes
Avoid using v_signmask universal intrinsic and mark it as deprecated

```
force_builders=Custom,Linux AVX2,ARMv7,ARMv8,Linux32,Win32
buildworker:Custom=linux-3
build_image:Custom=ubuntu:18.04
CPU_BASELINE:Custom=AVX512_SKX
build_image:Custom Win=msvs2019
CPU_BASELINE:Custom Win=AVX512_SKX
```

### CODE REVIEW

File: modules/core/include/opencv2/core/hal/intrin_cpp.hpp [hunk 1]
```diff
@@ -1088,6 +1089,23 @@ template<typename _Tp, int n> inline int v_signmask(const v_reg<_Tp, n>& a)
...
+/** @brief Get first negative lane index
+
+Returned value is an index of first negative lane (undefined for input of all positive values)
+Example:
+@code{.cpp}
+v_int32x4 r; // set to {0, 0, -1, -1}
+int idx = v_heading_zeros(r); // idx = 2
+@endcode
+*/
+template <typename _Tp, int n> inline int v_find_negative(const v_reg<_Tp, n>& a)
```
Comment (alalek) [inline modules/core/include/opencv2/core/hal/intrin_cpp.hpp]: What is about these names?
- `v_find_non_zero()`
- `v_scan_non_zero()`
- `v_scan_forward()`
- `v_scan()`

File: modules/core/include/opencv2/core/hal/intrin_neon.hpp [hunk 2]
```diff
@@ -1103,10 +1103,25 @@ inline int v_signmask(const v_uint64x2& a)
...
+inline int v_find_negative(const v_int8x16& a) { return trailingZeros32(v_signmask(a)); }
+inline int v_find_negative(const v_uint8x16& a) { return trailingZeros32(v_signmask(a)); }
+inline int v_find_negative(const v_int16x8& a) { return trailingZeros32(v_signmask(a)); }
+inline int v_find_negative(const v_uint16x8& a) { return trailingZeros32(v_signmask(a)); }
+inline int v_find_negative(const v_int32x4& a) { return trailingZeros32(v_signmask(a)); }
+inline int v_find_negative(const v_uint32x4& a) { return trailingZeros32(v_signmask(a)); }
+inline int v_find_negative(const v_float32x4& a) { return trailingZeros32(v_signmask(a)); }
+#if CV_SIMD128_64F
+inline int v_find_negative(const v_int64x2& a) { return trailingZeros32(v_signmask(a)); }
+inline int v_find_negative(const v_uint64x2& a) { return trailingZeros32(v_signmask(a)); }
```
Comment (alalek) [inline modules/core/include/opencv2/core/hal/intrin_neon.hpp]: It is strange that `CV_SIMD128_64F` blocks `int64` / `uint64` processing.

**Comment by ChipKerchner** [comment]
Why are we deprecating v_signmask?  It's a useful instruction.

**Comment by alalek** [comment]
There is "wide universal intrinsics (w.u.i.)" API over SIMD128 / SIMD256 / SIMD512 (depends on current compilation flags). So we are trying to unify APIs if possible (however it doesn't work well for reductions).

v_signmask() in case of SIMD512 doesn't fit well into 32-bit int (u8 x 64 elements).

**Comment by ChipKerchner** [comment]
BTW, IMO v_signmask should return an 'unsigned int' and not an 'int'

**Pull request merged**